lineapy/graph_reader/artifact_collection.py

```diff
@@ -42,55 +46,60 @@ def __init__(
-            self._get_artifacts_grouped_by_session(target_artifacts)
-        )
-
-        # Retrieve reuse precomputed artifact objects and group them by session ID
-        self.pre_computed_artifacts_by_session = (
-            self._get_artifacts_grouped_by_session(
-                reuse_pre_computed_artifacts
-            )
+        # Check if target artifacts name has been repeated in the input list
+        check_duplicates(target_artifacts)
+        # Check if entries in reuse pre computed artifacts list have been repeated
+        check_duplicates(reuse_pre_computed_artifacts)
+
+        # ------ Initialize Linea Artifacts from artifactdef ------
+        # Retrieve target artifact objects
+        self.target_artifacts: List[
+            LineaArtifact
+        ] = get_db_artifacts_from_artifactdef(self.db, target_artifacts)
+
+        # Retrieve reuse precomputed artifact objects
+        self.pre_computed_artifacts: List[
+            LineaArtifact
+        ] = get_db_artifacts_from_artifactdef(
+            self.db, reuse_pre_computed_artifacts
         )
 
+        # ------ Group artifacts by session using SessionArtifact ------
         # For each session, construct SessionArtifacts object
         self.session_artifacts: Dict[LineaID, SessionArtifacts] = {}
         for (
             session_id,
-            session_artifacts,
-        ) in self.target_artifacts_by_session.items():
-            # TODO: LIN-653, LIN-640 only collect matched pre_computed artifacts in this Session
-            # instead of grabbing all Artifacts as currently required by SessionArtifact
-            pre_calculated_artifacts: List[LineaArtifact] = sum(
-                self.pre_computed_artifacts_by_session.values(), []
-            )
+            target_artifacts_by_session,
+        ) in get_artifacts_grouped_by_session(self.target_artifacts):
 
             self.session_artifacts[session_id] = SessionArtifacts(
                 self.db,
-                session_artifacts,
+                target_artifacts_by_session,
                 input_parameters=input_parameters,
-                reuse_pre_computed_artifacts=pre_calculated_artifacts,
+                # TODO: LIN-653, LIN-640 only collect matched pre_computed artifacts in this Session
+                # instead of grabbing all Artifacts as currently required by SessionArtifact
+                reuse_pre_computed_artifacts=self.pre_computed_artifacts,
             )
 
         # TODO: LIN-653, LIN-640
         # Add some type of validation that all re-use compute artifacts can be matched to
         # a target artifact before the creation of SessionArtifacts.
+        # If calls to .values() are repeated in the class, it might indicate a need for a new datastructure to hold the values
+        all_sessions_artifacts = [
+            art.name
+            for sa in self.session_artifacts.values()
+            for art in sa.all_session_artifacts.values()
```

Comment:
another instance of confusing names here. session_artifacts has variable named all_session_artifacts. Since they arent used elsewhere in the class and because refactoring SessionArtifacts itself is out of scope, these have been marked for future work. 
The calls to .values() in my opinion is a flag however i've heard opinions against it as well so leaving this as is.